gap/oper.gi

```diff
@@ -1771,3 +1771,67 @@ function(digraph, k)
 
   return DigraphByAdjacencyMatrix(out);
 end);
+
+#
+
+InstallMethod(DIGRAPHS_IsMatching,
+"for a digraph and a list",
+[IsDigraph, IsHomogeneousList],
+function(gr, edges)
+  local seen, e;
+
+  if not IsDuplicateFreeList(edges)
+      or not ForAll(edges, e -> IsDigraphEdge(gr, e)) then
+    return false;
+  fi;
+
+  seen := BlistList(DigraphVertices(gr), []);
+
+  for e in edges do
+    if seen[e[1]] or seen[e[2]] then
+      return false;
+    fi;
+    seen[e[1]] := true;
+    seen[e[2]] := true;
+  od;
+
+  return seen;
+end);
+
+InstallMethod(IsMatching,
+"for a digraph and a list",
+[IsDigraph, IsHomogeneousList],
+function(gr, edges)
+  return DIGRAPHS_IsMatching(gr, edges) <> false;
+end);
+
+InstallMethod(IsPerfectMatching,
+"for a digraph and a list",
+[IsDigraph, IsHomogeneousList],
+function(gr, edges)
+  local seen;
+
+  seen := DIGRAPHS_IsMatching(gr, edges);
+  if seen = false then
+    return false;
+  fi;
+  return SizeBlist(seen) = DigraphNrVertices(gr);
+end);
+
+InstallMethod(IsMaximalMatching,
+"for a digraph and a list",
+[IsDigraph, IsHomogeneousList],
+function(gr, edges)
+  local seen, e;
+
+  seen := DIGRAPHS_IsMatching(gr, edges);
+  if seen = false then
+    return false;
+  fi;
+  for e in DigraphEdges(gr) do
```

Comment:
Instead of calling `DigraphEdges` (which can be costly) why not just loop through each list in the `OutNeighbours`?